activesupport/lib/active_support/number_helper/rounding_helper.rb

```diff
@@ -0,0 +1,64 @@
+module ActiveSupport
+  module NumberHelper
+    class RoundingHelper # :nodoc:
+      attr_reader :options
+
+      def initialize(options)
+        @options = options
+      end
+
+      def round(number)
+        return number unless precision
+        number = convert_to_decimal(number)
+        if significant && precision > 0
+          round_significant(number)
+        else
+          round_without_significant(number)
+        end
+      end
+
+      def digit_count(number)
+        return 1 if number.zero?
+        (Math.log10(absolute_number(number)) + 1).floor
+      end
+
+      private
+        def round_without_significant(number)
+          number = number.round(precision)
+          number = number.to_i if precision == 0 && number.finite?
+          number = number.abs if number.zero?
```

Comment:
Perhaps `number = 0 if number.zero?` is easier to understand?

Either way, I think this line is obscure enough that you should keep the `prevent showing negative zeros` comment.